.github/workflows/rust_build.yml

```diff
@@ -0,0 +1,17 @@
+name: rust_build
+
+on:
+    workflow_call
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: dtolnay/rust-toolchain@stable
+        with:
+          toolchain: stable
+      - run: rustup component add rustfmt clippy
```

Comment:
We should run clippy and fmt essentially a pre-commit checks as part of build pipeline

Reply:
This workflow is being called in the build workflow